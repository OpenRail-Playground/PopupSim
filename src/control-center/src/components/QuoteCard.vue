<script setup lang="ts">
import type { Quote } from '@/utils/api'

defineProps<{
  quote: Quote
}>()
</script>

<template>
  <div class="cmp-card quote-card">
    <h2 class="elm-headline">{{ quote.title }}</h2>
    <p>
      {{ quote.body }}
    </p>
    <div class="btn-group">
      <RouterLink
        :to="{ name: 'quote', params: { slug: quote.slug } }"
        class="elm-button"
        data-variant="secondary-outline"
        data-size="small"
        type="button"
        data-icon-after="arrow-forward"
      >
        {{ $t('show') }}
      </RouterLink>
      <RouterLink
        :to="{ name: 'editQuote', params: { slug: quote.slug } }"
        class="elm-button is-icon-text-replace"
        data-variant="tertiary-plain"
        data-size="small"
        type="button"
        data-icon-after="edit"
      >
        {{ $t('edit') }}
      </RouterLink>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.quote-card {
  padding: 1rem;

  .elm-headline {
    margin-top: 0;
  }

  .btn-group {
    justify-content: space-between;
  }
}
</style>
