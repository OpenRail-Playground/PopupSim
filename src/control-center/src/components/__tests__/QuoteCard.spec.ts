import { shallowMount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import QuoteCard from '../QuoteCard.vue'
import type { Quote } from '@/utils/api'
import { createI18n } from 'vue-i18n'

const quote: Quote = {
  id: 1,
  slug: 'quote-1',
  title: 'Quote 1',
  body: 'A great quote.',
  author: 'John Doe',
  notes: 'A short note',
  createdAt: '2023-09-27T09:19:19.042Z',
  updatedAt: '2023-10-12T18:36:03.630Z',
  owner: 'abcdef'
}

describe('QuoteCard', () => {
  test('renders the quote card', () => {
    const i18n = createI18n({ legacy: false })
    const wrapper = shallowMount(QuoteCard, {
      props: { quote },
      global: {
        plugins: [i18n],
        stubs: ['RouterLink']
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
