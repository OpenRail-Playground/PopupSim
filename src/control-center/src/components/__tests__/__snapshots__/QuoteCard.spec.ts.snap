// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuoteCard > renders the quote card 1`] = `
<div
  class="cmp-card quote-card"
  data-v-74e1d4f6=""
>
  <h2
    class="elm-headline"
    data-v-74e1d4f6=""
  >
    Quote 1
  </h2>
  <p
    data-v-74e1d4f6=""
  >
    A great quote.
  </p>
  <div
    class="btn-group"
    data-v-74e1d4f6=""
  >
    <router-link-stub
      class="elm-button"
      data-icon-after="arrow-forward"
      data-size="small"
      data-v-74e1d4f6=""
      data-variant="secondary-outline"
      to="[object Object]"
      type="button"
    />
    <router-link-stub
      class="elm-button is-icon-text-replace"
      data-icon-after="edit"
      data-size="small"
      data-v-74e1d4f6=""
      data-variant="tertiary-plain"
      to="[object Object]"
      type="button"
    />
  </div>
</div>
`;
